tests/integration_tests/modules/test_ntp_servers.py

```diff
@@ -28,31 +30,61 @@
       - 172.16.15.15
 """
 
+USER_DATA_UPDATE_TIMESYNCD = """\
+bootcmd:
+- apt-get update -y
+- apt-get install -y systemd-timesyncd
+"""
+
 EXPECTED_SERVERS = yaml.safe_load(USER_DATA)["ntp"]["servers"]
 EXPECTED_POOLS = yaml.safe_load(USER_DATA)["ntp"]["pools"]
 
 
-@pytest.mark.user_data(USER_DATA)
+def compose_user_data() -> str:
+    """Return applicable user-data for the current release.
+
+    Devel releases frequently see systemd* package updates
+    which introduce complex dependency resolver changes in the devel
+    series when trying to install ntp package due to stale
+    apt sources representations making it hard to resolve
+    packaging conflicts as both ntpsec and systemd-timesyncd
+    Provides: ntp
+
+    To avoid this inability to resolve conflicts we need to
+    ensure the installed systemd-timesyncd is updated to the latest
+    published version before trying to install ntpd.
+    Use bootcmd for this work because the cc_ntp module runs before
+    cc_package_update_upgrade_install so we can't add packages:[...]
+    """
+    if CURRENT_RELEASE == UBUNTU_DEVEL:
```

Comment:
Should we do this regardless of version? I've seen this fail on released noble too.